component/stream/read_write.go

```diff
@@ -203,25 +202,41 @@ func (rw *ReadWriteCache) RenameFile(options internal.RenameFileOptions) error {
 	err := rw.NextComponent().RenameFile(options)
 	if err != nil {
 		log.Err("Stream::RenameFile : error renaming file %s [%s]", options.Src, err.Error())
+	}
+	return err
+}
+
+func (rw *ReadWriteCache) FlushFile(options internal.FlushFileOptions) error {
```

Comment:
For direct streaming, data is already uploaded so we can just return back success here in that case.